src/OpenTelemetry/Metrics/MetricPoint/MetricPoint.cs

```diff
@@ -394,7 +394,7 @@ internal void Update(long number)
             case AggregationType.LongSumIncomingCumulative:
             case AggregationType.LongGauge:
                 {
-                    Interlocked.Exchange(ref this.runningValue.AsLong, number);
+                    Volatile.Write(ref this.runningValue.AsLong, number);
```

Comment:
curious, if this shows improvement in the metric stress tests?

Reply:
@pentp If there is a reasonable improvement in performance, I will spend time analyzing the complete metrics SDK to understand how these changes affect it.